worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/StorageManager.scala

```diff
@@ -416,6 +416,40 @@ final private[worker] class StorageManager(conf: CelebornConf, workerSource: Abs
     }.toSeq.sortBy(_._2).reverse.take(conf.metricsAppTopDiskUsageCount * 2).toMap.asJava
   }
 
+  def cleanFile(shuffleKey: String, fileName: String): Unit = {
+    val fileInfo = getFileInfo(shuffleKey, fileName)
+    if (fileInfo != null) {
+      cleanFileInternal(shuffleKey, fileInfo)
+    }
+  }
+
+  def cleanFileInternal(shuffleKey: String, fileInfo: FileInfo): Boolean = {
+    var isHdfsExpired = false
+    if (fileInfo.isHdfs) {
+      isHdfsExpired = true
+      val hdfsFileWriter = hdfsWriters.get(fileInfo.getFilePath)
+      if (hdfsFileWriter != null) {
+        hdfsFileWriter.destroy(new IOException(
+          s"Destroy FileWriter ${hdfsFileWriter} caused by shuffle ${shuffleKey} expired."))
+        hdfsWriters.remove(fileInfo.getFilePath)
```

Comment:
Will it happen that hdfsFileWriter is null but the directory is not empty? cc @zhongqiangczq  Why is old code written that way?

Reply:
"hdfsFileWriter is null but the directory is not empty" can't happen, because delete files first and then remove. "if (hdfsFileWriter != null)" can be removed

Reply:
> Will it happen that hdfsFileWriter is null but the directory is not empty? cc @zhongqiangczq Why is old code written that way?

Currently batch release thread may release partition while shuffle expired, and this can be very rare case.

Reply:
@RexXiong If it happens, will it cause problem?

Reply:
> @RexXiong If it happens, will it cause problem?

As cleanFile method is idempotent & support concurrent, So it would works fine even that scenario.